addons/sale_stock/models/replenishment_warhouse_select.py

```diff
@@ -0,0 +1,45 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class SaleStockReplenishmentWarehouseSelect(models.TransientModel):
+    _name = 'sale_stock.replenishment.warehouse.select'
+    _description = 'Replenishment Warehouse Select'
+    _rec_name = 'sale_order_line_id'
+
+    sale_order_line_id = fields.Many2one('sale.order.line')
+    product_id = fields.Many2one('product.product', related='sale_order_line_id.product_id')
+    qty_to_order = fields.Float(related='sale_order_line_id.product_uom_qty')
+
+    # TODO: here need to retrieve all warehouses?
+    route_ids = fields.One2many(related='sale_order_line_id.order_id.warehouse_id.resupply_route_ids')
```

Comment:
For this one (the wizard on the sale order line), we don't want to find the routes by looking on the warehouse resupply routes, but by checking in all the company warehouses, right?